### PR TITLE
feature-BH-6712: Remove retry and integrate with job. Update slack no…

### PR DESCRIPTION
**Pull request by mohammed-fahad**
…tification to use different body size for different attributes.

### CODE REVIEW

File: src/Job/Job.php [hunk 1]
```diff
@@ -2,22 +2,37 @@
...
+    protected $strategy;
+    protected $errorMessage;
+    protected $notify;
+
+    public function __construct(
+        Message $message,
+        int $delays,
+        int $maxRetryCount,
+        string $strategy,
+        string $errorMessage = '',
```
Comment (kaleongchan) [inline src/Job/Job.php:27]: I don't suppose you have an error message to set when creating a new job.

**Review by DannyCui — APPROVED** [review]

**Review by afareh — APPROVED** [review]

File: src/Job/Job.php [hunk 2]
```diff
@@ -93,4 +92,86 @@ public function getJson(): ?stdClass
...
+    {
+        return $this->strategy;
+    }
+
+    public function getErrorMessage()
+    {
+        return $this->errorMessage;
+    }
+
+    public function getNotify()
```
Comment (kaleongchan) [inline src/Job/Job.php:117]: How about `shouldNotify`?  Can we please have type hint?

**Review by kaleongchan — COMMENTED** [review]
Looks good overall. Just a couple of small things. :+1:

**Review by kaleongchan — APPROVED** [review]
:+1:

**Pull request merged**